agent/recorder-agent.c

```diff
@@ -0,0 +1,419 @@
+
+  g_debug ("hwangsae_recorder_agent_dbus_unregister");
+
+  if (self->manager)
+    g_dbus_interface_skeleton_unexport (G_DBUS_INTERFACE_SKELETON
+        (self->manager));
+
+  if (self->recorder_interface)
+    g_dbus_interface_skeleton_unexport (G_DBUS_INTERFACE_SKELETON
+        (self->recorder_interface));
+
+  g_application_release (app);
+
+  /* chain up */
+  G_APPLICATION_CLASS (hwangsae_recorder_agent_parent_class)->dbus_unregister
+      (app, connection, object_path);
+}
+
+gboolean
+    hwangsae_recorder_agent_recorder_interface_handle_start
+    (Hwangsae1DBusRecorderInterface * object,
+    GDBusMethodInvocation * invocation, gchar * arg_id, gpointer user_data) {
+  g_autofree gchar *cmd = NULL;
+  g_autofree gchar *response = NULL;
+  GError *error = NULL;
+  HwangsaeRecorderAgent *self = (HwangsaeRecorderAgent *) user_data;
+
+  g_debug ("hwangsae_recorder_agent_recorder_interface_handle_start, cmd %s",
+      cmd);
+
+  hwangsae_recorder_agent_start_recording (self, arg_id);
+
+  hwangsae1_dbus_recorder_interface_complete_start (object, invocation);
+
+  return TRUE;
+}
+
+gboolean
+    hwangsae_recorder_agent_recorder_interface_handle_stop
+    (Hwangsae1DBusRecorderInterface * object,
+    GDBusMethodInvocation * invocation, gchar * arg_id, gpointer user_data) {
+  g_autofree gchar *cmd = NULL;
+  g_autofree gchar *response = NULL;
+  GError *error = NULL;
+  HwangsaeRecorderAgent *self = (HwangsaeRecorderAgent *) user_data;
+
+  g_debug ("hwangsae_recorder_agent_recorder_interface_handle_stop, cmd %s",
+      cmd);
+
+  hwangsae_recorder_agent_stop_recording (self, arg_id);
+
+  hwangsae1_dbus_recorder_interface_complete_stop (object, invocation);
+
+  return TRUE;
+}
+
+static void
+hwangsae_recorder_agent_dispose (GObject * object)
+{
+  HwangsaeRecorderAgent *self = HWANGSAE_RECORDER_AGENT (object);
+
+  g_clear_object (&self->recorder);
+  g_clear_object (&self->settings);
+  g_clear_object (&self->manager);
+  g_clear_object (&self->recorder_interface);
+
+  G_OBJECT_CLASS (hwangsae_recorder_agent_parent_class)->dispose (object);
+}
+
+static void
+hwangsae_recorder_agent_class_init (HwangsaeRecorderAgentClass * klass)
+{
+  GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
+  GApplicationClass *app_class = G_APPLICATION_CLASS (klass);
+
+  gobject_class->dispose = hwangsae_recorder_agent_dispose;
+
+  app_class->dbus_register = hwangsae_recorder_agent_dbus_register;
+  app_class->dbus_unregister = hwangsae_recorder_agent_dbus_unregister;
+}
+
+static gboolean
+signal_handler (GApplication * app)
+{
+  g_application_release (app);
+
+  return G_SOURCE_REMOVE;
+}
+
+static void
+hwangsae_recorder_agent_init (HwangsaeRecorderAgent * self)
+{
+  gchar *uid = NULL;
+
+  self->is_recording = FALSE;
+  self->got_file_created_signal = FALSE;
+  self->got_file_completed_signal = FALSE;
+
+  self->settings = g_settings_new (HWANGSAE_RECORDER_SCHEMA_ID);
+
+  self->recorder = hwangsae_recorder_new ();
```

Comment:
Maybe you've agreed this with @justinjoy, but your agent supports only one recording at a time. You should create a new `HwangsaeRecorder` for each `Start` request.

Thinking a bit ahead now, we should eventually run each recorder as a separate process, so that crash of one doesn't bring down all of them.

Reply:
As I understand, for the demo it should be one recording at a time in two different machines.

> Thinking a bit ahead now, we should eventually run each recorder as a separate process, so that crash of one doesn't bring down all of them.

I'm sure about how @justinjoy would like to handle this, to let several `hwangsae-recorder-agent` we should agree on how to handle dbus name and applications flags, as currently only one instance is allowed to run.

Reply:
> As I understand, for the demo it should be one recording at a time in two different machines.

If that's the case, the current implementation should do for now.

> ... to let several hwangsae-recorder-agent we should agree on how to handle dbus name ...

What I have in mind is several recording processes, but only one `hwangsae-recorder-agent` managing them. That way DBus name clashes etc. become a non issue as there's always only one process holding those resources.

Reply:
Good point. TBH, I missed it. A recorder process should be able to handle multiple streams from edge, but for now, as @xhaakon said, let's keep the current implementation (per process recording), and estimate the task to support multiple streams later.